rhel_osv/csaf.py

```diff
@@ -0,0 +1,112 @@
+import json
+from pathlib import Path
+from typing import Any
+
+
+class Reference(object):
+    def __init__(self, csaf_ref: dict[str, str]):
+        self.category = csaf_ref["category"]
+        self.summary = csaf_ref["summary"]
+        self.url = csaf_ref["url"]
+
+
+class Remediation(object):
+    def __init__(self, csaf_product_id: str, cpes: dict[str, str], purls: dict[str, str]):
+        split_product = csaf_product_id.split(":", maxsplit=1)
+        split_product_len = len(split_product)
+        if split_product_len == 1:
+            raise ValueError(f"Did not find ':' in product_id: {csaf_product_id}")
+        self.product = split_product[0]
+        self.cpe = cpes.get(self.product)
+        component_version = split_product[1]
+        self.product_version = component_version
+        self.purl = purls.get(self.product_version)
+        split_component_version = component_version.rsplit("-", maxsplit=2)
+        if len(split_component_version) != 3:
+            raise ValueError(f"Could not convert component into NEVRA: {component_version}")
```

Comment:
What's a NEVRA? (I'll take my answer as a comment/docstring in the code 😉)

Reply:
I'll add a comment.